clang/lib/DPCT/ExprAnalysis.cpp

```diff
@@ -1147,13 +1146,13 @@ void ExprAnalysis::analyzeType(TypeLoc TL, const Expr *CSCE) {
 
   auto Iter = MapNames::TypeNamesMap.find(TyName);
   if (Iter != MapNames::TypeNamesMap.end()) {
-    HelperFeatureSet.insert(Iter->second->RequestFeature);
-    requestHelperFeatureForTypeNames(TyName, SR.getBegin());
+    NeedDpctDeviceExtFlag = true;
```

Comment:
Should use `DpctGlobalInfo::setNeedDpctDeviceExt();`?

Reply:
This `NeedDpctDeviceExtFlag` is a member in ExprAnalysis class, not the one in DpctGlobalInfo.
Current change just follows the original logic.